Update egress gateway docs (#1067)
tigera/docs: calico-cloud/networking/egress/egress-gateway-aws.mdx

1. The `maxNextHops` field specifies the maximum number of egress gateway replicas from the selected deployment that a pod depends on. For more information, see [Optimize egress networking for workloads with long-lived TCP connections](egress-gateway-maintenance.mdx#reduce-the-impact-of-gateway-downtime).

> 1. `gatewayPreference` allows to specify hints to the gateway selection process. Default is `None` which leads to default selection process. If set to `PreferNodeLocal` then egress gateways local to the client's node will be used if available. If there are no local egress gateways, {{prodname}} will fallback to using the other egress gateways. In this example, for the default route, egress gateways local to the client's node will be used if present. If not, all egress gateways matching the selector will be used.

Reviewer: ```suggestion
1. `gatewayPreference` specifies hints to the gateway selection process. The default `None`, selects the default selection process. If set to `PreferNodeLocal`, then egress gateways local to the client's node are used if available. If there are no local egress gateways, {{prodname}} uses other egress gateways. In this example, for the default route, egress gateways local to the client's node are used if present. If not, all egress gateways matching the selector are used.
```
